### PR TITLE
Add quick redux-form fixes

### PR DESCRIPTION
**Pull request by LKay**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "../tslint.json" }`.

### CODE REVIEW

File: types/redux-form/lib/reduxForm.d.ts [hunk 1]
```diff
@@ -34,7 +34,7 @@ export interface Config<FormData extends DataShape, P, S> {
      * the name of your form and the key to where your form's state will be
      * mounted under the redux-form reducer
      */
-    form: string;
+    form?: string;
```
Comment (yuit) [inline types/redux-form/lib/reduxForm.d.ts]: Q: in this [document ](http://redux-form.com/6.6.3/docs/api/ReduxForm.md/) it seems that form should be required? let me know if I misunderstand something here

**Comment by yuit** [comment]
@LKay  look good but just small question

File: types/redux-form/lib/reduxForm.d.ts (hunk 1 above)
Comment (LKay) [inline types/redux-form/lib/reduxForm.d.ts]: It is required but can be defined both as configured property or by passing a property directly to the component as it states in the docs: `IMPORTANT: All of these configuration options may be passed into reduxForm() at "design time" or passed in as props to your component at runtime.`.

**Comment by Grmiade** [comment]
Thanks @LKay for this PR ;)
Another fix, I think `input` should be required, no ?

```
interface WrappedFieldProps<S> {
    input: WrappedFieldInputProps;
    meta: WrappedFieldMetaProps<S>;
}
```

**Comment by Grmiade** [comment]
And `handleSubmit` shouldn't be optional too I think ;)

```
handleSubmit(event: SyntheticEvent<any>): void; // same as ReactEventHandler
handleSubmit(submit: SubmitHandler<FormData, P, S>): ReactEventHandler<any>;
```

**Comment by LKay** [comment]
@Grmiade

About first, technically yes, as it is injected property but was left like that for now because it's not breaking anything. The is need to refactor the code to review existing types and anddress such issues and I'll do it when I'm more available.

For the second thing, it should be optional. You don't need to pass custom handler and if you don't then pass `handleSubmit` as it is to have default `redux-form` behavior.

**Comment by Grmiade** [comment]
I don't think `handleSubmit` can be undefined. Its use is optional but it isn't optional.

This example raises an error `[ts] Object is possibly 'undefined'.` on `handleSubmit`.
```
<form onSubmit={handleSubmit(this.handleSubmit)}>
  {...}
</form>
```

I am beginner with Typescript, I maybe missing something :/

**Comment by LKay** [comment]
Ah, I get you now. I though you were referring to `hendleSubmit`'s call parameter. Yes, `handleSubmit` should be defined as not optional because it is injected property. Like I said, some refactoring on properties needs to be done so this will be most likely addressed then.

File: types/redux-form/lib/reduxForm.d.ts (hunk 1 above)
Comment (yuit) [inline types/redux-form/lib/reduxForm.d.ts]: Reading..... 😭

**Pull request merged**